themes/src/main/resources/theme/keycloak.v2/login/field.ftl

```diff
@@ -45,6 +45,7 @@
   <@group name=name label=label error=error required=required>
     <span class="${properties.kcInputClass} <#if error?has_content>${properties.kcError}</#if>">
         <input id="${name}" name="${name}" value="${value}" type="text" autocomplete="${autocomplete}" <#if autofocus>autofocus</#if>
+                <#if autocomplete == "one-time-code">inputmode="numeric" pattern="[0-9]*"</#if>
```

Comment:
Considering `inputmode` is [widely supported](https://caniuse.com/input-inputmode), the `pattern` fallback should not be needed here. Unless there is another reason to keep it, I think it should be removed.
```suggestion
                <#if autocomplete == "one-time-code">inputmode="numeric"</#if>
```